src/plugins/dashboard/public/dashboard_app/listing_page/get_dashboard_list_item_link.ts

```diff
@@ -12,27 +12,48 @@ import { IKbnUrlStateStorage, setStateToKbnUrl } from '@kbn/kibana-utils-plugin/
 
 import {
   DASHBOARD_APP_ID,
+  DASHBOARD_STATE_STORAGE_KEY,
   GLOBAL_STATE_STORAGE_KEY,
   createDashboardEditUrl,
 } from '../../dashboard_constants';
 import { coreServices } from '../../services/kibana_services';
+import { DASHBOARD_STATE_SESSION_KEY } from '../../services/dashboard_backup_service';
 
 export const getDashboardListItemLink = (
   kbnUrlStateStorage: IKbnUrlStateStorage,
   id: string,
-  timeRestore: boolean
+  timeRestore: boolean,
+  spaceId?: string
 ) => {
   const useHash = coreServices.uiSettings.get('state:storeInSessionStorage'); // use hash
 
   let url = coreServices.application.getUrlForApp(DASHBOARD_APP_ID, {
     path: `#${createDashboardEditUrl(id)}`,
   });
+
   const globalStateInUrl = kbnUrlStateStorage.get<QueryState>(GLOBAL_STATE_STORAGE_KEY) || {};
 
   if (timeRestore) {
     delete globalStateInUrl.time;
     delete globalStateInUrl.refreshInterval;
   }
   url = setStateToKbnUrl<QueryState>(GLOBAL_STATE_STORAGE_KEY, globalStateInUrl, { useHash }, url);
+  // pull the filters off the session storage and put in the url if they exist
+  const unsavedFiltersToUrl = JSON.parse(
+    sessionStorage.getItem(DASHBOARD_STATE_SESSION_KEY) ?? '[]'
+  );
+
+  const unsavedFilters = spaceId && unsavedFiltersToUrl ? unsavedFiltersToUrl[spaceId] : undefined;
```

Comment:
Should unsaved queries also be forwarded to be consistent?

Reply:
They should be, yes :)

Reply:
Maybe we can re-use the code that generates the share link so they are the same